legal-api/src/legal_api/services/document_meta.py

```diff
@@ -0,0 +1,169 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This provides the service for filings documents meta data."""
+
+from enum import Enum
+
+from legal_api.models import Filing
+from legal_api.utils.legislation_datetime import LegislationDatetime
+
+
+class DocumentMetaService():
+    """Provides service for document meta data."""
+
+    class DocumentType(Enum):
+        """Define an enum of document types."""
+
+        REPORT = 'REPORT'
+
+    class ReportType(Enum):
+        """Define an enum of report types."""
+
+        CERTIFICATE = 'certificate'
+        NOTICE_OF_ARTICLES = 'noa'
+
+    def get_documents(self, filing: dict):
+        """Return an array of document meta for a filing."""
+        business_identifier = filing['filing']['business']['identifier']
+        filing_id = filing['filing']['header']['filingId']
+        filing_date = filing['filing']['header']['date']
+        filing_status = filing['filing']['header']['status']
+        filing_type = filing['filing']['header']['name']
+        paper_only = filing['filing']['header']['availableOnPaperOnly']
+        documents = []
+
+        if filing_status not in (Filing.Status.COMPLETED.value, Filing.Status.PAID.value) or paper_only:
+            return []
+
+        if filing_type == 'incorporationApplication':
+            documents = self.get_incorporation_application_report(filing)
+            return documents
+
+        if filing_status == Filing.Status.COMPLETED.value:
+            if filing_type == 'annualReport':
+                documents = self.get_annual_report(filing_id, business_identifier, filing_date)
+            elif filing_type == 'changeOfAddress':
+                documents = self.get_coa_report(filing_id, business_identifier, filing_date)
+            elif filing_type == 'changeOfDirectors':
+                documents = self.get_cod_report(filing_id, business_identifier, filing_date)
+            elif filing_type == 'changeOfName':
+                documents = self.get_con_report(filing_id, business_identifier, filing_date)
+            elif filing_type == 'specialResolution':
+                documents = self.get_special_resolution_report(filing_id, business_identifier, filing_date)
+            elif filing_type == 'voluntaryDissolution':
+                documents = self.get_voluntary_dissolution_report(filing_id, business_identifier, filing_date)
+
+        return documents
+
+    def get_annual_report(self, filing_id, business_identifier, filing_date):
+        """Return an annual report document meta object."""
+        return [self.create_report_object(filing_id, 'Annual Report',
```

Comment:
Since all these are calling the same method with different arguments, is it worth to have a map kind of structure with the filing type as key and  an array of objects as value. Each of that object will have the file name, query param, is  draft version available as properties so that any future reports will have to keep adding the required data to the array and not touch the framework code? We can do it post mvp also. Just thinking whether that would be a nice thing to do.

Reply:
I understand that there is phase 2 for this ticket to provide the links. If my above comment makes sense, it can be done at that point.

Reply:
@lekshmimallika-aot I will take a look at this in another PR if that is okay. I agree we can probably do something a bit better here.